plugins/obs-ffmpeg/texture-amf.cpp

```diff
@@ -1384,10 +1437,77 @@ try {
 	return false;
 }
 
+static inline void check_recommended_avc_defaults(amf_base *enc,
```

Comment:
These functions should probably be called `set_` or `adjust_` rather than `check_`. The prefix `check_` implies that it's only doing settings value validation and allowing or stopping the init. What these functions actually do is change the settings values if "needed", even if it is only changing defaults, which should only apply if the user has not explicitly changed values.

Reply:
Fixed in [409550a](https://github.com/obsproject/obs-studio/pull/9352/commits/409550a6bc3a9b1250ca0a538b90e23cd6757f20)